packages/shared/lib/core/network/interfaces/network.interface.ts

```diff
@@ -6,8 +6,8 @@ export interface INetwork {
     getMetadata(): NetworkMetadata
     getStatus(): INetworkStatus
 
-    getChain(chainId: number): IChain
-    getChains(): IChain[]
+    getChain(chainId: number): IChain | undefined
+    getChains(): (IChain | undefined)[]
```

Comment:
remove `unefined` here if you agree on the above comment

Reply:
I agree, but for l.9 it's a bit difficult, since `find` can return `undefined`. Alternatively, we can throw an error if the chain isn't found, but I'm not sure if that should happen within the class itself.